operation.go

```diff
@@ -296,9 +307,14 @@ func (o *Operation) Run(handler interface{}) {
 		}
 
 		setFields(ctx, ctx.r, input, inputType)
+		if !ctx.HasError() {
+			// No errors yet, so any errors that come after should be treated as a
+			// semantic rather than structural error.
+			ctx.errorCodeHint = http.StatusUnprocessableEntity
+		}
 		resolveFields(ctx, "", input)
 		if ctx.HasError() {
-			ctx.WriteError(http.StatusBadRequest, "Error while parsing input parameters")
+			ctx.WriteError(ctx.errorCodeHint, "Error while processing input parameters")
```

Comment:
nit: Should this be `ctx.errorCode` rather than `ctx.errorCodeHint`?  It's directly written at the response, not hinting at it.